packages/sitecore-jss/src/layout-service.ts

```diff
@@ -0,0 +1,182 @@
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+import { AxiosDataFetcher } from './data-fetcher';
+import { LayoutServiceData } from './dataModels';
+import * as dataApi from './dataApi';
+import { HttpJsonFetcher } from './httpClientInterface';
+import { IncomingMessage, ServerResponse } from 'http';
+
+interface LayoutServiceConfig {
+  apiHost: string;
+  apiKey: string;
+  siteName: string;
+  /**
+   * Enables/disables analytics tracking for the Layout Service invocation (default is true).
+   * More than likely, this would be set to false for SSG/hybrid implementations, and the
+   * JSS tracker would instead be used on the client-side: {@link https://jss.sitecore.com/docs/fundamentals/services/tracking}
+   */
+  tracking?: boolean;
+}
+
+type FetchConfig = {
```

Comment:
Reading through everything, I think this `FetchConfig` has conflicting interests, and may not even be necessary. Seems like the properties here could be split up, moved to other locations more suited to their usage:

- `dataFetcher` - move to `LayoutServiceConfig`. There's no reason to be specified/created for each request
- `onReq`, `onReqError`, `onRes`, `onResError` - move to `AxiosDataFetcher` constructor. These are only used here anyway.
- `req`, `res` - move to `fetchLayoutData` as optional parameters (in place of current `FetchConfig`)

Reply:
Good idea, changed